plutus-core/cost-model/CostModelGeneration.md

```diff
@@ -150,9 +151,13 @@ codebase.
     a check that the inputs are the same length.  For more complicated functions
     one might also put the implementation in a separate file.
 
-    The final argument of `makeBuiltinMeaning` contains the costing functions for
-    the relevant builtin.  Initially this should be set to `mempty`; we'll come
-    back and fix it later.
+    The final argument of `makeBuiltinMeaning` contains the costing functions
+    for the relevant builtin.  Initially this should be set to
+    `runCostingFun<N>Arguments . def` (where `<N>` is the number of arguments
```

Comment:
I had no idea this would work! Just checked, it does work. I don't like it TBH, it looks too "normal", I'd rather have an explicit `maxBound` in there or something else that screams "this isn't implemented properly yet". Even just a dedicated type class would be helpful I think:

```haskell
class NotCostedYet model where
    notCostedYet :: BuiltinCostModel -> model
```

Reply:
OK, I'll think about that.

Reply:
>  I'd rather have an explicit maxBound in there or something else that screams "this isn't implemented properly yet"

I added a suitable typeclass with a method called `unimplementedCostingFun` that you're supposed to use until you've got a proper one.  Maybe I should have made it `costingFunUNIMPLEMENTED` or something.

Reply:
I couldn't make the cost of an "unimplemented" costing function `maxBound` because then the evaluator just fails all the time, so I made it ten times the current maximum on-chain CPU limit, which should be safe.  I tried to remove the `def` instances, but it turns out that without them you get an error [here](https://github.com/IntersectMBO/plutus/blob/9ec15b09238cb139b105a1c79f0f346a3a3350a9/plutus-core/testlib/PlutusIR/Test.hs#L191) because `compileWIthOpts` requires them (I'm not quite sure why).  It looks as if it copes OK if the default values are all `maxBound` (they were previously 0), so I've left them like that to be safe.

Reply:
We could add a hlint warning for it!

Reply:
> We could add a hlint warning for it!

![image](https://github.com/IntersectMBO/plutus/assets/10480926/a9c71667-fc9c-42a9-9be3-b5ab48fce842)